# **Practical Task 1**
Run object detection on any image.

In [2]:
from ultralytics import YOLO
from PIL import Image
import matplotlib.pyplot as plt

In [4]:
# Load YOLOv8 model
model = YOLO("yolov8n.pt")

# Run object detection on an image
results = model("image.jpg", save=True)

print("Object detection completed!")




image 1/1 D:\AI-ML Internship\Day46\image.jpg: 448x640 2 cats, 239.0ms
Speed: 11.7ms preprocess, 239.0ms inference, 5.2ms postprocess per image at shape (1, 3, 448, 640)
Results saved to D:\AI-ML Internship\Day46\runs\detect\predict-5
Object detection completed!


In [7]:
for result in results:
    result.show()

# **Practical Task 2**
Write down: <br>
• Detected Objects  
• Confidence Scores 

| Detected Object | Confidence Score |
| --------------- | ---------------- |
| Cat             | 91%              |
| Cat             | 80%              |


# **Practical Task 3**
Experiment with: <br>
conf=0.3 <br>
 
conf=0.5 <br>
 
conf=0.8 <br>
Observe differences

In [9]:
model = YOLO("yolov8n.pt")

# Confidence = 0.3
results_03 = model("image.jpg", conf=0.3, save=True)

# Confidence = 0.5
results_05 = model("image.jpg", conf=0.5, save=True)

# Confidence = 0.8
results_08 = model("image.jpg", conf=0.8, save=True)


image 1/1 D:\AI-ML Internship\Day46\image.jpg: 448x640 2 cats, 192.6ms
Speed: 9.1ms preprocess, 192.6ms inference, 3.9ms postprocess per image at shape (1, 3, 448, 640)
Results saved to D:\AI-ML Internship\Day46\runs\detect\predict-6

image 1/1 D:\AI-ML Internship\Day46\image.jpg: 448x640 2 cats, 192.2ms
Speed: 10.2ms preprocess, 192.2ms inference, 3.5ms postprocess per image at shape (1, 3, 448, 640)
Results saved to D:\AI-ML Internship\Day46\runs\detect\predict-6

image 1/1 D:\AI-ML Internship\Day46\image.jpg: 448x640 2 cats, 203.6ms
Speed: 11.5ms preprocess, 203.6ms inference, 3.9ms postprocess per image at shape (1, 3, 448, 640)
Results saved to D:\AI-ML Internship\Day46\runs\detect\predict-6


##### observations
| Confidence Threshold | Detected Objects |
| -------------------- | ---------------- |
| 0.3                  | 2 Cats           |
| 0.5                  | 2 Cats           |
| 0.8                  | 2 Cats           |


# **Practical Task 4**
Print: <br>
result.boxes<br> 
and analyze output.

In [11]:
for result in results:
    print(result.boxes)

ultralytics.engine.results.Boxes object with attributes:

cls: tensor([15., 15.])
conf: tensor([0.9082, 0.8005])
data: tensor([[0.0000e+00, 1.3925e+03, 2.2554e+03, 2.6128e+03, 9.0819e-01, 1.5000e+01],
        [2.1618e+03, 1.3247e+02, 3.4514e+03, 2.6361e+03, 8.0053e-01, 1.5000e+01]])
id: None
is_track: False
orig_shape: (2948, 4423)
shape: torch.Size([2, 6])
xywh: tensor([[1127.7076, 2002.6868, 2255.4153, 1220.2882],
        [2806.5950, 1384.2925, 1289.6250, 2503.6494]])
xywhn: tensor([[0.2550, 0.6793, 0.5099, 0.4139],
        [0.6345, 0.4696, 0.2916, 0.8493]])
xyxy: tensor([[   0.0000, 1392.5426, 2255.4153, 2612.8308],
        [2161.7825,  132.4678, 3451.4075, 2636.1172]])
xyxyn: tensor([[0.0000, 0.4724, 0.5099, 0.8863],
        [0.4888, 0.0449, 0.7803, 0.8942]])


##### output analyze
| Attribute  | Meaning                             |
| ---------- | ----------------------------------- |
| cls        | Two detected objects, both Cats     |
| conf       | 90.82% and 80.05% confidence        |
| xyxy       | Bounding box coordinates            |
| xyxyn      | Normalized bounding box coordinates |
| orig_shape | Original image size                 |


# **Practical Task 5**
Identify: <br>
• Object Class  <br>
• Confidence  <br>
• Bounding Box  <br>
for every detected object.

| Object   | Class | Confidence | Bounding Box (xyxy)                         |
| -------- | ----- | ---------- | ------------------------------------------- |
| Object 1 | Cat   | 90.82%     | [0.0000, 1392.5426, 2255.4153, 2612.8308]   |
| Object 2 | Cat   | 80.05%     | [2161.7825, 132.4678, 3451.4075, 2636.1172] |
